core-primitives/stf-interface/src/lib.rs

```diff
@@ -0,0 +1,71 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+//! Provides a state interface.
+//! This allow to easily mock the stf and exchange it with another storage.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+use itp_types::OpaqueCall;
+
+#[cfg(feature = "mocks")]
+pub mod mocks;
+pub mod parentchain_pallet;
+pub mod sudo_pallet;
+pub mod system_pallet;
+
+pub trait StateInterface<State, StateDiff> {
+	fn init_state(&self, initial_input: Vec<u8>) -> State;
```

Comment:
I think `initial_input` is misleading as a name, it is the encoded enclave account ID. `initial_input` would suggest it is an initial state that is set, but that's not the case.

Reply:
The initial idea was to not enforce the input being an account ID. It's the implementation's task to define what the input should be. Is it not?

Reply:
In this case, there is no room for interpretation or generalization. The fact that this is the encoded enclave account, is not an implementation detail, but part of the interface.

Reply:
Hm. Alright, conceded 👍